### PR TITLE
Update README and CHANGELOG about updates 3.5.1

### PR DESCRIPTION
**Pull request by feliperenan**
These updates are about using Simple Form with non Active Record objects.

More information: https://github.com/plataformatec/simple_form/issues/1549

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1049,6 +1049,66 @@ by passing the html5 option:
...
+end
+```
+
+You can define all methods required by the helpers.
+
+```ruby
+class User 
+  extend ActiveModel::Naming
+
+  def id; end;
```
Comment (davidbrusius) [inline README.md]: We don't need the semicolon after the `end` ✂️. There are other points we have to fix this, take a look at them too. 😉

File: README.md [hunk 2]
```diff
@@ -1049,6 +1049,66 @@ by passing the html5 option:
...
+end
+```
+
+If you are using Presenters or Decorators that inherit from `SimpleDelegator` you can delegate 
+it to the model.
+
+```ruby
+class UserPresenter < SimpleDelegator
+  # Without that, Simple Form will consider the user model as the object.
+  def to_model; self; end
```
Comment (davidbrusius) [inline README.md]: Just a suggestion:

```ruby
def to_model
  self
end
```

Would it be more readable if we go like this? 

I tend to use single line definition when the method body is empty, but it's just my preference.

File: README.md [hunk 3]
```diff
@@ -1049,6 +1049,76 @@ by passing the html5 option:
...
+    id 
+  end
+
+  def persisted? 
+    false 
+  end
+end
+```
+
+If your object don't implement those methods, you must explicit it when you are building
```
Comment (tegon) [inline README.md]: you must explicit it **tell it** when you are building the form

File: README.md [hunk 4]
```diff
@@ -1049,6 +1049,76 @@ by passing the html5 option:
...
+```
+
+If your object don't implement those methods, you must explicit it when you are building 
+the form.
+
+```ruby
+class User 
+  attr_accessor :id, :name
+
+  # The only method required for use f.submit helper.
```
Comment (tegon) [inline README.md]: The only method required **to** use **the** f.submit helper.

**Pull request merged**